modules/social_features/social_album/social_album.module

```diff
@@ -562,3 +563,30 @@ function social_album_social_core_compatible_content_forms_alter(&$compatible_co
   $compatible_content_type_forms[] = 'node_album_form';
   $compatible_content_type_forms[] = 'node_album_edit_form';
 }
+
+/**
+ * Implements hook_node_links_alter().
+ */
+function social_album_node_links_alter(array &$links, NodeInterface $node, array &$context): void {
+  // 'Edit' button on current Album page for album creator and administrator.
+  if ($context['view_mode'] === 'full' && $node->getType() === 'album') {
+    $album_id = $node->id();
+    $current_user = \Drupal::currentUser();
+    $album_creator_id = $node->getOwnerId();
+    $link_edit_album = [
+      'title' => 'Edit',
+      'url' => Url::fromRoute('entity.node.edit_form', ['node' => $album_id]),
+      'weight' => '0',
+    ];
+
+    if (
+      (
+        $current_user->hasPermission('edit own album content') &&
+        $current_user->id() === $album_creator_id
```

Comment:
Can we just use `$node->access('edit', $current_user)` here instead?

Reply:
This solution doesn't work

Reply:
Interesting why, because usually it should work.
Then at least better to replace `in_array('administrator', $current_user->getRoles())` with check for `administer nodes` permission.

Reply:
**administer nodes** permission also not really correct permission since this permission also is enabled for CM, SM. In this [Jira comment](https://getopensocial.atlassian.net/browse/PROD-27132?focusedCommentId=142723) I described how edit works on teasers